File: java/src/civitas/SorpresaJugadorEspeculador.java
package civitas;

public class SorpresaJugadorEspeculador extends Sorpresa {
    int fianza;
    Jugador jugador;

    SorpresaJugadorEspeculador(Jugador jugador, int fianza) {
        super("Sorpresa Jugador Especulador");
        this.fianza = fianza;
        this.jugador = jugador;
    }

    public void aplicarAJugador_salirCarcel(int actual, ArrayList<Jugador> todos) {
        jugador = new JugadorEspeculador(todos.get(actual), fianza);
        
    }
}

File: ruby/lib/sorpresa_jugadorespeculador.rb
require_relative ('jugador_especulador')

module Civitas
    class SorpresaJugadorEspeculador < Sorpresa
        public_class_method :new
        def initialize(fianza)
            super("Jugador Especulador")
            @valor=fianza
        end

        def aplicarAJugador(actual, todos)
            if(jugadorCorrecto(actual, todos))
                informe(actual, todos)
                jugador=JugadorEspeculador.nuevoEspeculador(todos[actual], @valor)
                todos.delete_at(actual)
                todos.insert(actual, jugador)
            end
        end

    end
end